### PR TITLE
Update WakaTime.Shared.ExtensionUtils dependency to v4.2.1

### LINKED ISSUE #23: MS Office Add-in API Key Formatting
The current implementation of WakaTime MS Office add-ins do not accept the API key in [waka__***...] format. In order for the API key to be accepted by the Office add-in, the "waka__" prefix needs to be stripped leaving the rest of the API key.

### LINKED ISSUE #24: Not tracking activity
i insalled word plugin and enter api-key. but it not added to my dashboard.

### LINKED ISSUE #29: `Please enter valid Api Key.` when trying to change to new wakatime key
image
The key is copied directly from wakatime and put into excel without any proxies... Still I get the error `Please enter valid Api Key.`

### LINKED ISSUE #30: New cli version available
I have this message on my Plugin Status dashboard. I tried to re-install the plugin for excel as states the instructions but it didn't work, I also tried to manually install the cli; I did, but it changed nothing on the status. The release for office is from Jul 4, 2021, but the one for the cli is from yesterday (Aug 30, 2023), is this related to the issue?

image
image

### PR DESCRIPTION
**Pull request by Jozoooooo**
- Update WakaTime.Shared.ExtensionUtils dependency to v4.2.1
- Adapt to new `WakaTime API Key` format
- Fix #23
- Fix #24
- Fix #29
- Fix #30

### CODE REVIEW

**Comment by Jozoooooo** [comment]
Hello @gandarez @alanhamlett , I recently discovered the problem that the `WakaTime.Shared.ExtensionUtils v2.1.1` is too old to be used when using `office-wakatime`, so I updated the `WakaTime.Shared.ExtensionUtils v4.2.1` and made some optimizations. The effect is as shown below. The official website's "Plugin Status" can correctly list Word, Excel, and PPT and can be used normally. I think #23, #24, #29, and #30 have been fixed.
Snipaste

**Comment by Jozoooooo** [comment]
My Release version is linked below https://github.com/Jozoooooo/office-wakatime/releases/tag/4.2.1.

**Comment by ApolloLSY** [comment]
Yesterday I encountered the same problem as #29 in Word.
Now I have uninstalled the old version and installed version 4.2.1. But it still not working on my computer.
2eb00e1dfb39249633f9586eb670347
bf478af8c4fa58cb7ada0efbce75d51
1e4a7d36ec8ec579e4adcdfc76005cb

**Comment by Jozoooooo** [comment]
Thanks for your feedback @ApolloLSY ! In order to resolve the problem, I need you to provide the log file `word-wakatime.log` by these following steps:
1. Turn on the `Debug` mode in `WakaTime Settings` window.
2. Open a word document normally, then edit and save it.
3. Find `C:\Users\{YOURUSERNAME}\AppData\Roaming\WakaTime\word-wakatime.log`. 
4. Upload the log file.

**Comment by ApolloLSY** [comment]
[word-wakatime.log](https://github.com/wakatime/office-wakatime/files/13883382/word-wakatime.log)

I just found that this version is running successfully now. It seems it was a delay issue...  ×w×
Sorry for previous mistaken report.
Thank you for your attention!

**Comment by Jozoooooo** [comment]
You're welcome! 😃

**Comment by alanhamlett** [comment]
A lot of other optimizations along with the version bump. @gandarez can you help review this?

**Comment by gandarez** [comment]
I'll review this PR and see what's missing and test locally then.

**Comment by Jozoooooo** [comment]
Hello @gandarez @alanhamlett ! During the process of modifying the code, I found that there were many libraries that were not used in the code, so I tried to delete them, and no problems were found in my local environment. Thank you both for your replies. 😄

File: ExcelWakaTimeAddin/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -33,6 +33,6 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("2.1.1")]
-[assembly: AssemblyFileVersion("2.1.1")]
+[assembly: AssemblyVersion("4.2.1")]
```
Comment (alanhamlett) [inline ExcelWakaTimeAddin/Properties/AssemblyInfo.cs]: We don't want to bump the extension version. This is different from the `WakaTime.Shared.ExtensionUtils` version.

File: ExcelWakaTimeAddin/WakaTimeAddIn.cs [hunk 2]
```diff
@@ -57,7 +61,7 @@ private static void ApplicationOnWorkbookActivate(Excel.Workbook wb)
         {
             try
             {
-                WakaTime.HandleActivity(wb.FullName, false, string.Empty);
+                WakaTime.HandleActivity(wb.FullName, false, "Microsoft Office");
```
Comment (alanhamlett) [inline ExcelWakaTimeAddin/WakaTimeAddIn.cs]: We should leave the project empty and let `wakatime-cli` detect it, or let the user set [custom rules](https://wakatime.com/faq#custom-rules) to define projects.

**Comment by Jozoooooo** [comment]
Thank you for maintaining this project, I’m glad to be able to contribute to it. 😄

**Pull request merged**